libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -506,3 +534,97 @@ def test_that_case_sensitivity_does_not_affect_distance_strategy(
     conn.execute(text("use master"))
     conn.execute(text(_DROP_COLLATION_DB_QUERY))
     conn.close()
+
+
+@pytest.mark.parametrize("test_filter, expected_ids", TYPE_1_FILTERING_TEST_CASES)
+def test_sqlserver_with_with_metadata_filters_1(
+    store: SQLServer_VectorStore,
+    test_filter: Dict[str, Any],
+    expected_ids: List[int],
+) -> None:
+    store.add_texts(filter_texts, filter_metadatas, filter_ids)
+
+    docs = store.similarity_search("meow", k=5, filter=test_filter)
+    store.delete(["1", "2", "3"])
```

Comment:
is delete called to cleanup the store? If yes, better to do it after the assert for logical order in the tests:

# init

# act

# cleanup

Reply:
good point, can there be something like a finally clause - which does the cleanup and is executed no matter if the test fails or succeeds?

Reply:
Yes, added.